index.d.ts

```diff
@@ -0,0 +1,63 @@
+declare module "moleculer-apollo-server" {
+	import { ServiceSchema } from "moleculer";
+	import { Config } from "apollo-server-core";
+	export {
+		GraphQLUpload,
+		GraphQLExtension,
+		gql,
+		ApolloError,
+		toApolloError,
+		SyntaxError,
+		ValidationError,
+		AuthenticationError,
+		ForbiddenError,
+		UserInputError,
+		defaultPlaygroundOptions,
+	} from "apollo-server-core";
+
+	export * from "graphql-tools";
+
+	export interface ApolloServerOptions {
+		path: string;
+		disableHealthCheck: boolean;
+		onHealthCheck: () => {};
+	}
+
+	export class ApolloServer {
+		createGraphQLServerOptions(req: any, res: any): Promise<any>;
+		createHandler(options: ApolloServerOptions): void;
+		supportsUploads(): boolean;
+		supportsSubscriptions(): boolean;
+	}
+
+	export interface GraphQLNodeResolver {
+		action: string;
+		rootParams?: {
+			[key: string]: string;
+		};
+		dataloader?: boolean;
+	}
+
+	export interface GraphQLTypeResolver {
+		[key: string]: {
+			[key: string]: GraphQLNodeResolver;
+		};
+	}
+
+	export interface ApolloServiceOptions {
+		typeDefs?: string;
+		resolvers?: GraphQLTypeResolver;
```

Comment:
I'm getting errors from this type for every use we have currently on the global `resolvers`.

There definitely is a facility to make "action" resolvers through here, but you can make many other resolver types which this is not accomodating for.

In our current project, we do not have any "action" resolvers on the global definition.  We have [scalar resolvers](https://www.apollographql.com/docs/graphql-tools/scalars/#custom-scalars):
```
import { GraphQLDate, GraphQLDateTime, GraphQLTime } from 'graphql-iso-date';
...
resolvers: {
  Date: GraphQLDate,
  Time: GraphQLTime,
  DateTime: GraphQLDateTime,
```

We have [enumeration resolvers](https://www.apollographql.com/docs/graphql-tools/scalars/#enums):
```
enum PhoneType {
	home = '1',
	work = '2',
	mobile = '3',
}
...
resolvers: {
  ...
  PhoneType: {
    home: PhoneType.home,
    work: PhoneType.work,
    mobile: PhoneType.mobile,
  },
  ...
```

We have [abstract resolvers](https://www.apollographql.com/docs/apollo-server/schema/unions-interfaces/#interface-type):
```
const Item = {
	__resolveType(item: Package | Product): string | null {
		if (isPackage(item)) {
			return 'Package';
		}

		if (isProduct(item)) {
			return 'Product';
		}

		return null;
	},
};
...
resolvers: {
  ...
  Item,
  ...
```

The current type definition covers the very "moleculer" specific usage of resolvers, but it doesn't cover the other use cases of resolvers.

Reply:
`makeExecuteableSchema` defines the acceptable types as `IResolvers<any, TContext> | Array<IResolvers<any, TContext>>`.  moleculer-apollo-server is converting the action resolver schema into this format, but if you don't provide an action resolver schema then the other types should just flow through.

If you make this a union of those types with the action resolver schema type (`GraphQLTypeResolver | IResolvers | Array<IResolvers>`) then I think this is fixed.

Reply:
Thanks. 👍